Flappy Bird Game Mode

heriom opened the pull request.

heriom committed:
Added Flappy Bird GameMode

Joshix-1 committed:
few changes

- use GameManager.end_game() to make the game integrate better
- use delta in movement calculations
- follow style guidelines

Numenter committed:
revert unrelated files

heriom committed:
Merge pull request #1 from Numenter/test_revert_bird

revert unrelated files

heriom committed:
Merge pull request #2 from Joshix-1/flappybird

few changes

Numenter commented:
The problem with the bird image persists.
#31

heriom commented:
if someone is willing to create a bird or smth. like that i'm ready to change it, so that the person just needs to create a new sprite, but as i've already told in #31 i am not an artist and really not talented.

ASecondGuy commented:
Artist and talented are relative. Just make something and it'll be good enough.

Joshix-1 commented:
I've tried my "best" :)
![bird](https://user-images.githubusercontent.com/57299889/172021788-8c98802f-b47a-4780-9c9c-7fcde6fdb096.png)

heriom committed:
Changed Bird Sprite

heriom committed:
Merge branch 'flappybird' of https://github.com/heriom/Suffragium into flappybird

heriom committed:
Merge branch 'flappybird'

Numenter commented on game/games/flappybird/Player.gd:
```diff
@@ -0,0 +1,68 @@
+# spawn a new wall at the right side of the screen
+func spawn_wall():
+	var instance = wall.instance()
+	rand_seed(rand_range(0, 25))
```
```suggestion
```
remove this

heriom replied:
thanks for the feedback.

i actually added line 47 because when running the game the normal rand_range created the same pattern over and over again. maybe it was another problem in the wall implementation but it was a quick fix.

i'll try to remove the line while fixing the problem

Joshix-1 replied:
You could do it like the testgame and use your own RandomNumberGenerator:
```gdscript
onready var _rng := RandomNumberGenerator.new()
```
and call `_rng.randomize()`

heriom replied:
i fixed it. thanks!

Numenter commented on game/games/flappybird/Player.gd:
```diff
@@ -0,0 +1,68 @@
+func _on_Hitbox_area_entered(area):
+	if area.name == "PointHitbox":
+		score += 1
+		print(score)
```
```suggestion
```
debug print?

heriom replied:
forgot that after testing. thanks!

Numenter reviewed (commented):
- Start the game on key input and add a control info box.
- I don't like that you get thrown out of the game when you lose. It should show an endscreen, with buttons to restart and leave.
- Wall and background graphics are missing
- Add a start, flap and finish sound?

Joshix-1 commented:
> I don't like that you get thrown out of the game when you lose. It should show an endscreen, with buttons to restart and leave.

The end screen should be done by the game manager and not by the game. That's something that'll be added in the future

> Start the game on key input and add a control info box.

Not needed games shouldn't do that. Stuff like that should be done by the GameManager. It wouldn't be good if every game implements their own menu.

ASecondGuy commented on game/games/flappybird/game.cfg:
```diff
@@ -0,0 +1,15 @@
+
+[game]
+; Choose a memorable name
+name ="Flappy Bird"
```
```suggestion
name="Flappy Bird"
```
The repo will be haunted by this typo forever...

heriom replied:
i will add some, thanks for the feedback!

ASecondGuy commented on game/games/flappybird/Player.gd:
```diff
@@ -0,0 +1,68 @@
+
+	motion = move_and_slide(delta * 60 * motion, UP)
+
+	get_parent().get_parent().get_node("CanvasLayer/RichTextLabel").text = str(score)
```
Declare an onready var like this 
`onready var _score_label := $"../../CanvasLayer/RichTextLabel"`
and then change this line to 
`_score_label.text = str(score)`.

It is bad practice to use get_node() every frame.

heriom replied:
makes sense. gonna add this

ASecondGuy reviewed (changes requested):
Please add a few more graphics and sounds. They don't need to be high quality or professionally made. Just make some and that'll be enough.

Numenter commented:
> > Start the game on key input and add a control info box.
> 
> Not needed games shouldn't do that. Stuff like that should be done by the GameManager. It wouldn't be good if every game implements their own menu.
![flapflap](https://user-images.githubusercontent.com/84922125/172046667-70a349a6-6d0f-4856-9902-a462331208b5.png)

Joshix-1 commented:
That isn't too bad, but I still think it would be better if the GameManager does menus. Then it is more uniform

heriom commented:
maybe we should ask this question everyone?

heriom committed:
Graphics and Sounds + Fixing Randomize Bug

fixed the randomize bug with the help of a script own RandomNumberGenerator

added some new graphics. background and new walls, yes they look pretty like the normal flappy bird but are still changable.

the game now has sounds and i added a sound when the player dies for later when the gamemanager handles the end of a game better

now the game starts with the press of a button and i changed how the game handles the start of a game

heriom committed:
Merge branch 'flappybird'

Joshix-1 commented:
did you make the graphics and sounds on your own?

heriom commented:
everything except the background. i am currently making a new one.

heriom committed:
New Background

heriom committed:
Merge branch 'flappybird'

Numenter committed:
new background images

Numenter commented:
quick pixeld Background
https://github.com/heriom/Suffragium/pull/4

heriom committed:
Merge pull request #4 from Numenter/flappy-background

new background image

Numenter commented:
Move audio and image files in subfolders.
I like to use:
```
/assets/images
/assets/sounds/sfx
```

Numenter commented:
Use snake_case for file names.
https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html#naming-conventions

Numenter commented:
Use sanke_case for file names.
https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_styleguide.html#naming-conventions

heriom committed:
New Folder Structure

Added new folders to make the structure more structured

Also changed file names to snake_case

heriom committed:
Merge branch 'flappybird'

heriom committed:
changed position of the bird

heriom committed:
Merge branch 'flappybird'

heriom committed:
Merge branch 'main' into main
